pkg/apis/core/validation/validation.go

```diff
@@ -2110,6 +2110,33 @@ func validateDataSource(dataSource *core.TypedLocalObjectReference, fldPath *fie
 	return allErrs
 }
 
+// validateDataSourceRef validates a DataSourceRef in a PersistentVolumeClaimSpec
+func validateDataSourceRef(dataSourceRef *core.TypedObjectReference, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if len(dataSourceRef.Name) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("name"), ""))
+	}
+	if len(dataSourceRef.Kind) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("kind"), ""))
+	}
+	apiGroup := ""
+	if dataSourceRef.APIGroup != nil {
+		apiGroup = *dataSourceRef.APIGroup
```

Comment:
I have a question out to api reviewers if we have validation on the APIGroup today. I found the validation logic for [CRDs here](https://github.com/kubernetes/apiextensions-apiserver/blob/470c040912b9e650acae65a8890bb180f933d59c/pkg/apis/apiextensions/validation/validation.go#L226) but haven't found the logic for built-ins.

If so, then we would have to potentially add ratcheting validation for the APIGroup since this logic already exists for the beta data populators feature.

Reply:
Looks like the apigroup validation is covered [here](https://github.com/kubernetes/kubernetes/blob/f01c9e8683adacbfbad58e5153dfac9ebf954c4b/staging/src/k8s.io/kube-aggregator/pkg/apis/apiregistration/validation/validation.go#L49-L57), so it would be good that we match the same validation rules. Since this is a gap in the existing feature and not introduced by this one, and that it can only result in invalid objects, I am ok if we followup on this in a separate PR.

The idea behind ratcheting validation is that we disallow invalid new objects, but we allow updates on existing objects if the old object is already invalid. This is an example of adding ratcheting validation: https://github.com/kubernetes/kubernetes/pull/81870

Reply:
Opened https://github.com/kubernetes/kubernetes/issues/113798 to track this